api/management-model/src/test/java/org/apache/polaris/core/admin/model/CatalogSerializationTest.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.admin.model;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Test suite for Catalog JSON serialization and deserialization.
+ *
+ * <p>Coverage includes:
+ *
+ * <ul>
+ *   <li>Basic serialization/deserialization of Catalog objects
+ *   <li>Handling of null and empty fields
+ *   <li>Special character handling in field values
+ *   <li>Unicode character support
+ *   <li>Whitespace preservation
+ *   <li>AWS role ARN validation
+ * </ul>
+ *
+ * Error handling coverage:
+ *
+ * <ul>
+ *   <li>Invalid JSON input
+ *   <li>Malformed JSON structure
+ *   <li>Invalid enum values
+ *   <li>Edge cases like very long catalog names
+ * </ul>
```

Comment:
nit: I'd also remove this javadoc. Test methods should self-describing and keeping test javadoc in sync with test code is overhead.